docs/src/Dropdown.doc.js

```diff
@@ -0,0 +1,642 @@
+        required: true,
+        description: 'Callback when the user selects an item',
+        href: 'default',
+      },
+      {
+        name: 'href',
+        type: 'string',
+        description:
+          'When supplied, wraps the item in a Link, and directs users to the url when item is selected.',
+        href: 'default',
+      },
+    ]}
+  />
+);
+
+card(
+  <PropTable
+    Component={Dropdown.Section}
+    name="Dropdown.Section"
+    id="Dropdown.Section"
+    props={[
+      {
+        name: 'children',
+        type: 'React.Node',
+        required: true,
+        description: 'Any Dropdown.Items to be rendered',
+        href: 'withSections',
+      },
+      {
+        name: 'label',
+        type: 'string',
+        required: true,
+        description: 'Label for the section',
+        href: 'withSections',
+      },
+    ]}
+  />
+);
+
+card(
+  <Card
+    description={`
+    <b>Important</b>: remember to include the following ARIA attributes on the element used for the \`anchor\` prop:
+
+    * \`aria-haspopup\` lets the screen reader know that there is a Dropdown menu linked to the trigger.
+    * \`aria-expanded\` informs the screen reader whether the Dropdown menu is currently open or closed.
+
+    Expected keyboard interaction:
+
+    * Hitting \`Enter\` or \`Space\` key on the Dropdown's trigger opens the menu
+    * \`Escape\` key closes the menu, while moving focus back on the Dropdown's trigger
+    * Arrow keys are used to navigate items within the menu
+    * \`Enter\` key selects an item within the Menu
+    * \`Tab\` or \` Shift + Tab\` close the menu and move focus accordingly
+  `}
+    name="Accessibility"
+  />
+);
+
+card(
+  <Example
+    id="default"
+    name="Basic Example"
+    description={`
+      A Dropdown can be attached to a [Button](/Button), [IconButton](/IconButton) or Textfield (via [Typeahead](/Typeahead)) using the \`anchor\` prop and is composed of multiple Dropdown Items.
+
+      If an item navigates to a new page, rather than acting as a selection, an \`href\` should be provided. If the item navigates to a page outside of the app, the \`isExternal\` prop should also be specified. Additionally, each item can contain a badge by specifying \`badgeText\`.
+
+      Dropdowns have a minimum width of \`180px\` and a maximum width of \`360px\`, after which the labels will truncate if needed. It will automatically open in the most convenient location, unless an \`idealDirection\` is specified.
+
+    `}
+    defaultCode={`
+      function DefaultMenuButtonDropdownExample() {
+        const [open, setOpen] = React.useState(false);
+        const [selected, setSelected] = React.useState(null);
+        const anchorRef = React.useRef(null);
+        const handleSelect = ({item}) => {
+          setSelected(item);
+        };
+
+        return (
+          <Box display="flex" justifyContent="center">
+            <Button
+              accessibilityHaspopup
```

Comment:
For future versions: is there a way to better encapsulate this behavior? Perhaps the dev has to use a "Dropdown.Button" component that automatically adds these props?

Reply:
I was thinking about this too - maybe a wrapper component, like you said Dropdown.Anchor or Dropdown.Trigger, that injects those props automatically? I'll add it to the ticket for the next round of updates